### ISSUE TITLE
A Volume Control

### ISSUE THREAD

**Issue by FaeyUmbrea** [bigbluebutton/bigbluebutton#12455]
One of the worst things about BigBlueButton is people yelling into your ear with painfully overtuned microphones while other people are almost inaudible because they happen to have sensible audio setups. 
I understand that per participant volume control is not feasible due to the stream being mixed on server and then delivered to the user. I am not asking for that.
I want to have a Mic and Output volume for my client specifically, so I can make my microphone a bit louder or quieter if others in the conference tell me my mic is to loud/quiet and I would like to be able to control the Volume of the BBB Application seperate from my entire browser.

Sidenote on possible workarounds that may pop up as suggestions:
Modifying Browser Volume is unacceptable because it would also affect other in browser audio sources (i.e. External Video Embeds)
Modifying Speaker Volume is unacceptable because it would also affect other Programs on the users machine
Modifying microphone volume in windows is unacceptable because it will upset all audio settings made to other software that has a volume control of the microphone (i.e. Skype, Teams, Discord) 

BBB is heavily in use especially by Universities due to Coronavirus and the experience is Tainted for people who use their PC for more than BBB and Office work due to the absense of this feature.

**Comment by jibon57** [comment]
Similar: #12194

**Comment by axkibe** [comment]
It should really be done in BBB. While technically you have a control in your operating system and on Linux (GNOME) this is trivial, in Windows 10 it is well hidden, and many Windows users don't seem to know how to do it. And if you ask them reduce their microphone boost, they change their Speaker Volume and ask if it is better..

AFAIK it should be possible from WebRTC to adjust microphone levels:
https://stackoverflow.com/questions/16942773/webrtc-is-it-possible-to-control-the-microphone-and-volume-levels

**Comment by Rosuav** [comment]
Even on Linux, there's a lot of value in having a volume control on BBB. It's not easy to balance volumes across different web sites without constantly adjusting things. Being forced to change my system volume, or the volume of all web browser processes, is not sufficient granularity to take advantage of everything.

I would very much like to be able to switch all my videoconferencing to BBB, and this is one of very few blockers.

**Comment by ffdixon** [comment]
We haven't had a chance to look deeper into this yet, but it remains on our road map as an enhancement to BigBlueButton's audio.

**Comment by FreeScholar** [comment]
Great work on version 2.4 looking forward to installing the newly released 2.5 next month!!!  

Any progress on this audio enhancement?I just got 2 emails from people asking about this very thing:)

**Comment by Rosuav** [comment]
Is this a planned enhancement? Autoclosing something just because people aren't talking about it seems dubious.

**Comment by FaeyUmbrea** [comment]
> Autoclosing something just because people aren't talking about it seems dubious.

This is literally a bot whose only function is to mark and close tickets with no activity

**Comment by Rosuav** [comment]
> > Autoclosing something just because people aren't talking about it seems dubious.
> 
> This is literally a bot whose only function is to mark and close tickets with no activity

Yeah, and it seems unnecessary that we have to make otherwise-meaningless posts just to say "hey bot, it's okay, this is still important to us". I've seen these kinds of "auto-close after X days" systems in a lot of places, and I dislike it in all of them.

Although if metrics are more important than code quality, then they do at least let you claim that 99% of issues have been successfully closed.

**Comment by FaeyUmbrea** [comment]
> I dislike it in all of them

Can we please keep this on topic?

**Comment by hostbbb** [comment]
![image](https://user-images.githubusercontent.com/519542/226173483-76d3778f-e2b2-4f34-a354-b39da467bef6.png)
This has been implemented in a firefox plugin here. https://github.com/Jo0001/BigBlueButtonPlus

Works with 2.6 RC8.... basically detects the talking events and adjust volume to the user set level.

Could be added in html5 client as inline javascript, and made to work with chrome and firefox...     Quick solution.  sure it could inspire a better written native implementation.

Thanks @Jo0001

**Comment by emoxam** [comment]
well ?

**Comment by Rosuav** [comment]
It would still be a great feature. Does this mean that the entire project is stale?

**Comment by ffdixon** [comment]
We have an automated bot that marks issues that have received no updates for 6 months as stale.  It's a way of culling out issues that are no longer relevant and gives the core maintainers a chance to review before they are automatically closed a few weeks later.

> Does this mean that the entire project is stale?

The BigBlueButton project is [very active](https://github.com/bigbluebutton/bigbluebutton/pulse/monthly).

**Comment by Rosuav** [comment]
Ah, good, an actual response from an actual person. It would be nice if people could either move this forward or close it, rather than having the bot staleify it, then it gets one response, then the bot unstales it, forever. Is this something that will happen or not?

**Comment by ffdixon** [comment]
There are about 200 people that have so far committed to the BigBlueButton project, and over 100 of them have done five commits or more.  So there are a lot of hands at work here.  Your question is also priority -- we focus pretty heavily on each release and its stabilization.    If a feature is part of a release, it gets implemented and stabilized.  If not, it must wait until a future release.

This request is a good suggestion, and we don't want to forget it (until someone suggests it anew).  

The answer to how a feature gets into a release is best explained [here](https://docs.bigbluebutton.org/support/faq/#when-will-feature-x-be-implemented).  

Looking ahead, we have a plug-in architecture in BigBlueButton 3.0, which will enable us to lower the effort to implement more features as we don't have to modify the core code.  It also enables our community to contribute to BigBlueButton without having to change the core as well. 

All to say, we're hopeful that features such as a volume control are not too far in the future for our community.
